### PR TITLE
Enhance network segregation

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

enhancement

## What does this PR do?

- put radicale and webmail on their own network: this is done for security: that way they have no privileged access anywhere (no access to redis, no access to XCLIENT, ...)
- remove the EXPOSE statements from the dockerfiles. These ports are for internal comms and are not meant to be exposed in any way to the outside world.

### Related issue(s)
- #2611

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [ ] In case of feature or enhancement: documentation updated accordingly
- [ ] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

File: webmails/roundcube/config/config.inc.php [hunk 1]
```diff
@@ -28,7 +28,7 @@
 $config['default_port'] = '10143';
 
 // Sieve script management
-$config['managesieve_host'] = '{{ IMAP_ADDRESS or "imap" }}';
+$config['managesieve_host'] = '{% if TLS_FLAVOR != "notls" %}ssl://{% endif %}{{ HOSTNAMES.split(",")[0] }}:4190';
```
Comment (ghostwheel42) [inline webmails/roundcube/config/config.inc.php:31]: What do we gain by connection through the public address here?
imap and smtp is done through front (unencrypted), so we could also du sieve through front unencrypted (port 14190)
Another (maybe better) way would be to do everything the webmails do through the public and encrypted endpoints like every client does.

File: core/nginx/conf/nginx.conf [hunk 2]
```diff
@@ -275,6 +277,24 @@ http {
     include /etc/nginx/conf.d/*.conf;
 }
 
+stream {
+    # managesieve with implicit TLS.
+    # we can't do explicit since dovecot doesn't have the certificates
+    # It's arguable whether we should use the standardized port number for doing something non-standard
```
Comment (ghostwheel42) [inline core/nginx/conf/nginx.conf:283]: I think it's fine. Clients should show an error.
Other proxies can detect TLS on the fly, so we could add starttls when a proxy is implemented.

Comment (nextgens) [reply core/nginx/conf/nginx.conf:289]: OK cool.

File: webmails/roundcube/config/config.inc.php (hunk 1 above)
Comment (nextgens) [reply webmails/roundcube/config/config.inc.php:31]: It bypasses the rate-limits and authentication is handled differently (with temporary tokens). Using the public ports would mean allowing that type of tokens on those ports and finding another way to rate-limit/exempt from rate limits.

Comment (nextgens) [reply webmails/roundcube/config/config.inc.php]: OK. Let's keep it simple.

**Comment by nextgens** [comment]
bors try

**Review by ghostwheel42 — APPROVED** [review]
lgtm

**Comment by ghostwheel42** [comment]
bors try

File: core/nginx/Dockerfile [hunk 3]
```diff
@@ -17,15 +17,16 @@ ARG VERSION
...
+  ; rm /etc/nginx/conf.d/stream.conf
 
 COPY conf/ /conf/
 COPY --from=static /static/ /static/
 COPY *.py /
 
 RUN echo $VERSION >/version
 
-EXPOSE 80/tcp 443/tcp 110/tcp 143/tcp 465/tcp 587/tcp 993/tcp 995/tcp 25/tcp 10025/tcp 10143/tcp
+EXPOSE 80/tcp 443/tcp 110/tcp 143/tcp 465/tcp 587/tcp 993/tcp 995/tcp 25/tcp 14190/tcp 10025/tcp 10143/tcp
```
[the remark this answers is not in the document]
Comment (nextgens) [reply core/nginx/Dockerfile]: This is an old revision

**Pull request merged**